src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -275,141 +235,38 @@ private void placeBlock(IBlockState localState, @NotNull Block localBlock, @NotN
-        {
-            for (int y = minY; y <= maxY; y++)
-            {
-                for (int z = minZ; z <= maxZ; z++)
-                {
-                    @NotNull BlockPos worldPos = new BlockPos(x, y, z);
-                    BlockPos localPos = worldPos.subtract(minPos);
-
-                    IBlockState blockState = world.getBlockState(worldPos);
-                    schematic.setBlockState(localPos, blockState);
-
-                    if (blockState.getBlock() instanceof AbstractBlockHut)
-                    {
-                        if (BlockPosUtil.isEqual(offset, 0, 0, 0))
-                        {
-                            BlockPosUtil.set(offset, localPos);
-                        }
-                        else
-                        {
-                            schematic.setBlockState(localPos, Blocks.air.getDefaultState());
-                            Log.getLogger().warn("Scan contained multiple AbstractBlockHut's ignoring this one");
-                        }
-                    }
-
-                    TileEntity tileEntity = world.getTileEntity(worldPos);
-                    saveTileEntity(schematic, tileEntity, minPos);
-                }
-            }
-        }
-        if (BlockPosUtil.isEqual(offset, 0, 0, 0))
-        {
-            offset.set(width / TWO_FOR_HALVING, 0, length / TWO_FOR_HALVING);
-        }
-        schematic.setOffset(offset);
-
-        @NotNull AxisAlignedBB region = AxisAlignedBB.fromBounds(minX, minY, minZ, maxX, maxY, maxZ);
-        //schematic::addEntity already does null checking and ignores null values.
-        world.getEntitiesWithinAABB(EntityHanging.class, region).forEach(schematic::addEntity);
-        world.getEntitiesWithinAABB(EntityMinecart.class, region).forEach(schematic::addEntity);
-
-        return schematic;
-    }
-
-    @NotNull
-    private static File getScanDirectory(@NotNull World world)
-    {
-        File minecolonies;
-        if (world.isRemote)
-        {
-            minecolonies = new File(Minecraft.getMinecraft().mcDataDir, "minecolonies/");
-        }
-        else
-        {
-            minecolonies = MinecraftServer.getServer().getFile("minecolonies/");
-        }
-        checkDirectory(minecolonies);
+        BlockPos blockpos =
+                new BlockPos(Math.min(from.getX(), to.getX()), Math.min(from.getY(), to.getY()), Math.min(from.getZ(), to.getZ()));
+        BlockPos blockpos1 =
+                new BlockPos(Math.max(from.getX(), to.getX()), Math.max(from.getY(), to.getY()), Math.max(from.getZ(), to.getZ()));
+        BlockPos size = blockpos1.subtract(blockpos).add(1, 1, 1);
 
-        @NotNull File scans = new File(minecolonies, "scans/");
-        checkDirectory(scans);
-        return scans;
-    }
+        WorldServer worldserver = (WorldServer) world;
+        MinecraftServer minecraftserver = world.getMinecraftServer();
+        TemplateManager templatemanager = worldserver.getStructureTemplateManager();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'theDir' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)